userspace/libsinsp/test/public_sinsp_API/events_set.cpp

```diff
@@ -125,19 +135,83 @@ TEST(events_set, names_to_event_set)
-	ASSERT_TRUE(event_codes.contains(PPME_SYSCALL_OPENAT2_E));
-	ASSERT_TRUE(event_codes.contains(PPME_SYSCALL_OPENAT2_X));
-	ASSERT_TRUE(event_codes.contains(PPME_SYSCALL_EXECVEAT_E));
-	ASSERT_TRUE(event_codes.contains(PPME_SYSCALL_EXECVEAT_X));
-	ASSERT_FALSE(event_codes.contains(PPME_GENERIC_E));
-	ASSERT_FALSE(event_codes.contains(PPME_GENERIC_X));
+	static libsinsp::events::set<ppm_event_code> event_codes_truth = {PPME_SYSCALL_OPENAT2_E, PPME_SYSCALL_OPENAT2_X,
+	PPME_SYSCALL_EXECVEAT_E, PPME_SYSCALL_EXECVEAT_X};
+	ASSERT_PPM_EVENT_CODES_EQ(event_codes, event_codes_truth);
+	ASSERT_TRUE(event_codes.equals(event_codes_truth));
 	ASSERT_EQ(event_codes.size(), 4); // enter/exit events for each event name
 
 	// Now insert a syscall bound to a generic event
 	event_names.insert("syncfs");
 	event_codes = libsinsp::events::names_to_event_set(event_names);
-	ASSERT_TRUE(event_codes.contains(PPME_GENERIC_E));
-	ASSERT_TRUE(event_codes.contains(PPME_GENERIC_X));
+	event_codes_truth = {PPME_SYSCALL_OPENAT2_E, PPME_SYSCALL_OPENAT2_X,
+	PPME_SYSCALL_EXECVEAT_E, PPME_SYSCALL_EXECVEAT_X, PPME_GENERIC_E, PPME_GENERIC_X};
+	ASSERT_PPM_EVENT_CODES_EQ(event_codes, event_codes_truth);
+	ASSERT_TRUE(event_codes.equals(event_codes_truth));
 	ASSERT_EQ(event_codes.size(), 6); // enter/exit events for each event name
+}
+
+TEST(events_set, event_set_to_names)
+{
+	static std::set<std::string> orderd_events_names_matching_set = {"kill", "dup"};
+	static libsinsp::events::set<ppm_event_code> events_set = {PPME_SYSCALL_KILL_E, PPME_SYSCALL_KILL_X,
+	PPME_SYSCALL_DUP_1_E, PPME_SYSCALL_DUP_1_X};
+	// TODO fix test for PPME_GENERIC_E, PPME_GENERIC_X after fixing method in another PR
+	auto events_names_final_set = libsinsp::events::event_set_to_names(events_set);
+	auto ordered_events_names_final_set = test_utils::unordered_set_to_ordered(events_names_final_set);
+	ASSERT_NAMES_EQ(ordered_events_names_final_set, orderd_events_names_matching_set);
+	ASSERT_EQ(events_names_final_set.size(), orderd_events_names_matching_set.size());
+}
 
+TEST(events_set, sc_set_to_event_set)
+{
+	libsinsp::events::set<ppm_sc_code> sc_set = {
+#ifdef __NR_kill
+	PPM_SC_KILL,
+#endif
+
+#ifdef __NR_sendto
+	PPM_SC_SENDTO,
+#endif
+
+// TODO discuss
+// #ifdef __NR_setresuid32
+	// PPM_SC_SETRESUID32,
+// #endif
+
+#ifdef __NR_alarm
+	PPM_SC_ALARM,
+#endif
+	};
+
+	libsinsp::events::set<ppm_event_code> event_set = {
+#ifdef __NR_kill
+	PPME_SYSCALL_KILL_E,
+	PPME_SYSCALL_KILL_X,
+#endif
+
+#ifdef __NR_sendto
+	PPME_SOCKET_SENDTO_E,
+	PPME_SOCKET_SENDTO_X,
+#endif
+
+// #ifdef __NR_setresuid32
+	// PPME_SYSCALL_SETRESUID_E,
+	// PPME_SYSCALL_SETRESUID_X,
+// #endif
+
+#ifdef __NR_alarm
+	PPME_GENERIC_E,
+	PPME_GENERIC_X,
+#endif
+	};
+
+	auto final_event_set = libsinsp::events::sc_set_to_event_set(sc_set);
+	ASSERT_PPM_EVENT_CODES_EQ(final_event_set, event_set);
+	ASSERT_TRUE(final_event_set.equals(event_set));
+}
+
+// TODO add
```

Comment:
Do you want to add it here or in another PR?

Reply:
yes seems better suited for a future PR after we clarified a few things and outstanding refactors are completed.